splitter.py

```diff
@@ -0,0 +1,147 @@
+"""Program to generate new split of a given dataset.
+
+This file contains functions required to generate new splits of a given dataset.
+Additionally, the new split generated will have the same distribution (intents)
```

Comment:
The key should be intents+slots.
Also, please add code to write both parts of the split (30-70) because you'd want to use the 70 separately for federated averaging later.

Reply:
Could you elaborate a bit on what you mean by "the key should be intent+slots"?

Reply:
As discussed offline, we want the dataset to have the same distribution on intent+slots. Let's do that if needed in a follow up PR.